ec2/auto_scaling_group/templates/scripts/asg_user_data.sh

```
@@ -0,0 +1,369 @@
+log() {
+    echo "[$(date '+%Y-%m-%d %H:%M:%S')] $1" | tee -a /var/log/asg-user-data.log
+}
+
+log "Auto Scaling Group User Data Script started"
+
+# ==================================================
+# AWS基本設定
+# ==================================================
+
+# AWS リージョンの決定（テンプレート変数が空の場合はメタデータから取得）
+AWS_REGION="${aws_region}"
```

Shellcheck (0.10.0)

[warning] aws_region is referenced but not assigned (did you mean 'AWS_REGION'?).

(SC2154)